apps/src/code-studio/pd/workshop_dashboard/attendance/session_attendance.jsx

```diff
@@ -2,39 +2,163 @@
  * Display and edit attendance for a workshop session, for display in a WorkshopAttendance tab.
  */
 import React from 'react';
+import $ from 'jquery';
+import _ from 'lodash';
 import SessionAttendanceRow from './session_attendance_row';
+import VisibilitySensor from '../components/visibility_sensor';
 import {Table} from 'react-bootstrap';
+import IdleTimer from 'react-idle-timer';
+
+// in milliseconds
+const REFRESH_DELAY = 5000;
+const IDLE_TIMEOUT = 60000;
+
+const styles = {
+  idle: {
+    opacity: .5
+  }
+};
 
 const SessionAttendance = React.createClass({
   propTypes: {
-    sessionId: React.PropTypes.number,
-    attendance: React.PropTypes.array.isRequired,
+    workshopId: React.PropTypes.number.isRequired,
+    sessionId: React.PropTypes.number.isRequired,
     adminOverride: React.PropTypes.bool,
-    onChange: React.PropTypes.func,
-    isReadOnly: React.PropTypes.bool
+    isReadOnly: React.PropTypes.bool,
+    onSaving: React.PropTypes.func.isRequired,
+    onSaved: React.PropTypes.func.isRequired
+  },
+
+  getInitialState() {
+    return {
+      loading: true,
+      attendance: undefined,
+      refreshInterval: undefined
```

Comment:
Why have this default to undefined? You can default it to a constant and save the if statement on line 50 and 57. Though if you absolutely want to wait for some more state to be set before calling the refreshing then it makes sense